docs/source/en/model_doc/patchtst.md

```diff
@@ -0,0 +1,74 @@
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# PatchTST
+
+## Overview
+
+The PatchTST model was proposed in [A Time Series is Worth 64 Words: Long-term Forecasting with Transformers](https://arxiv.org/abs/2211.14730) by Yuqi Nie, Nam H. Nguyen, Phanwadee Sinthong, Jayant Kalagnanam.
+
+The abstract from the paper is the following:
+
+*We propose an efficient design of Transformer-based models for multivariate time series forecasting and self-supervised representation learning. It is based on two key components: (i) segmentation of time series into subseries-level patches which are served as input tokens to Transformer; (ii) channel-independence where each channel contains a single univariate time series that shares the same embedding and Transformer weights across all the series. Patching design naturally has three-fold benefit: local semantic information is retained in the embedding; computation and memory usage of the attention maps are quadratically reduced given the same look-back window; and the model can attend longer history. Our channel-independent patch time series Transformer (PatchTST) can improve the long-term forecasting accuracy significantly when compared with that of SOTA Transformer-based models. We also apply our model to self-supervised pre-training tasks and attain excellent fine-tuning performance, which outperforms supervised training on large datasets. Transferring of masked pre-trained representation on one dataset to others also produces SOTA forecasting accuracy.*
+
+Tips:
+
+The model also adds a time series classification pipeline and time series regression pipeline.
+
+This model was contributed by [namctin](https://huggingface.co/namctin), [gsinthong](https://huggingface.co/gsinthong), [diepi](https://huggingface.co/diepi), [vijaye12](https://huggingface.co/vijaye12), [wmgifford](https://huggingface.co/wmgifford), and [kashif](https://huggingface.co/kashif).
+
+The original code can be found [here](https://github.com/yuqinie98/PatchTST).
+
+
+## PatchTSTConfig
+
+[[autodoc]] PatchTSTConfig
+
+
+## PatchTSTModel
+
+[[autodoc]] PatchTSTModel
+    - forward
+
+
+## PatchTSTForPrediction
+
+[[autodoc]] PatchTSTForPrediction
+    - forward
+
+
+## PatchTSTForForecasting
+
+[[autodoc]] PatchTSTForForecasting
+    - forward
```

Comment:
What's the difference between ForPrediction and ForForecasting?

Reply:
ForPrediction class is for the model that predicts the future values of one or few variables from a history of some other variables (predicting the future values of y given the input X). On the other hand, ForForecasting class, the model predicts the future values of X given its history.

Reply:
Thanks for the explanation, however would it be possible to just have a single `ForPrediction` class which can be used for both?

Reply:
I think it would be better to have two separate classes to distinguish them since each task requires its own way of defining the head.

Reply:
Ok, however the current naming is not consistent with the models we already have (Informer, AutoFormer, TimeSeriesTransformer). All of those have `xxxForPrediction` head models which predict future values given past values of the same variables (could be multiple in case of multivariate forecasting).

So it seems that ForForecasting here should actually be called ForPrediction, and ForPrediction needs to be called something else like ForOtherVariablePrediction (it's a bad name I know, cc @kashif for better suggestions)